packages/qwik-city/buildtime/vite/dev-server.ts

```diff
@@ -94,7 +99,20 @@ export function ssrDevMiddleware(ctx: BuildContext, server: ViteDevServer) {
         }
       }
 
-      // static file does not exist, 404
```

Comment:
current 404 path throw error. guest here's next goes to `qwik/optimizer/vite`'s plugin render. then dev404Middleware never trigger. IMHO that 404 handler put into previous code block is ok

Reply:
oh, 404 might need to be after staticDistMiddleware. (if not checking dist ouput filepath prefix)